core/src/main/scala/caliban/parsing/SourceMapper.scala

```diff
@@ -0,0 +1,41 @@
+package caliban.parsing
+
+import caliban.parsing.adt.LocationInfo
+import fastparse.internal.Util
+
+/**
+ * Maps an index to the "friendly" version of an index based on the underlying source.
+ */
+trait SourceMapper {
+
+  def getLocation(index: Int): LocationInfo
+
+}
+
+object SourceMapper {
+
+  /**
+   * Implementation taken from https://github.com/lihaoyi/fastparse/blob/master/fastparse/src/fastparse/ParserInput.scala
+   */
```

Comment:
you may need to reproduce his MIT license somewhere if you have not taken care of that.

Reply:
Good point. Is that usually at the file level or do we just need to have it once at the top level?

Reply:
I know of firms who do it at product level. You'd have to ask ZIO principals or Pierre I guess.

Reply:
I think in the file would be fine. Found an example in ZIO here https://github.com/zio/zio/blob/master/stacktracer/jvm/src/main/scala/zio/internal/stacktracer/impl/AkkaLineNumbers.scala